core/dbt/events/base_types.py

```diff
@@ -88,6 +101,29 @@ def get_pid(self) -> int:
             self.pid = os.getpid()
         return self.pid
 
+    def get_node_info(self):
+        return None
+
+
+@dataclass  #type: ignore
+class NodeInfo(Event, metaclass=ABCMeta):
+    report_node_data: Any
+    # node_status: str
+    # run_state: str
+
+    def get_node_info(self):
+        node_info = Node(
+            node_path=self.report_node_data.path,
+            node_name=self.report_node_data.name,
+            resource_type=self.report_node_data.resource_type,
+            materialized=self.report_node_data.config.materialized,
+            node_started_at="TODO",
+            unique_id=self.report_node_data.unique_id,
+            node_finished_at="TODO",
+            node_status="TODO"  # self.node_status  # pull from contract
+        )
```

Comment:
the fields you're accessing on `report_node_data` like `path`, `name`, `resource_type`, `config.materialized` etc. ARE the type that we've defined with the type hint `Any` above. I suspect we can represent this with a union of node types instead.